apps/malseasonal/mal_seasonal.star

```diff
@@ -0,0 +1,199 @@
+                                padding = 1,
+                                child = render.Row(
+                                    children = render_anime_titles(MAL_DATA),
+                                ),
+                            ),
+                        ),
+                    ],
+                ),
+            ],
+        ),
+        show_full_animation = True,
+    )
+
+def get_data(url, token):
+    res = http.get(url, headers = {
+        "Authorization": "Bearer %s" % token,
+    })
+    if res.status_code != 200:
+        fail("GET %s failed with status %d: %s", url, res.status_code, res.body())
+    return res.json()
+
+def oauth_handler(params):
+    params = json.decode(params)
+    res = http.post(
+        url = "https://myanimelist.net/v1/oauth2/token",
+        headers = {
+            "Accept": "application/json",
+        },
+        form_body = dict(
+            params,
+            code_verifier = CODE_VERIFIER,
+            client_secret = CLIENT_SECRET,
+        ),
+        form_encoding = "application/x-www-form-urlencoded",
+    )
+    if res.status_code != 200:
+        fail("token request failed with status code: %d - %s" %
+             (res.status_code, res.body()))
+
+    token_params = res.json()
+    access_token = token_params["access_token"]
+
+    return access_token
+
+def render_anime_titles(MAL_DATA):
+    animes = []
+
+    for anime in MAL_DATA["data"]:
+        animes.append(
+            render.Row(
+                main_align = "start",
+                children = [
+                    render.Image(
+                        src = http.get(anime["node"]["main_picture"]["medium"]).body(),
+                        width = 20,
+                        height = 25,
+                    ),
+                    render.Box(
+                        child = render.WrappedText(
+                            content = anime["node"]["title"],
+                            color = "#ffffff",
+                            align = "start",
+                            font = "CG-pixel-3x5-mono",
+                            linespacing = 1,
+                        ),
+                        width = 50,
+                        height = 25,
+                    ),
+                ],
+            ),
+        )
+
+    return animes
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.OAuth2(
+                id = "auth",
+                name = "MyAnimeList",
+                desc = "Login with MyAnimeList for access to your anime list.",
+                icon = "",
+                handler = oauth_handler,
+                client_id = "30323b16a7aed07ddccc0a3bf19ad8ef",
+                authorization_endpoint = "https://myanimelist.net/v1/oauth2/authorize?code_challenge=%s&client_id=%s&scope=read:user&redirect_uri=http://127.0.0.1:8080/oauth-callback&response_type=code&state=abc123" % (CODE_VERIFIER, CLIENT_ID),
```

Comment:
This should be considered for future implementations. There is no way right now to add other parameters like `code_challenge` for PKCE. This is required by some api's and it would be cool if it could be added to the Schema OAuth2 module.